python/lsst/cp/pipe/cpFunctions.py

```diff
@@ -0,0 +1,682 @@
+                    doc=self.inputScales.doc,
+                    storageClass=self.inputScales.storageClass,
+                    dimensions=self.allConnections['inputScales'].dimensions + newDimensions
+                )
+                self.dimensions.update(config.calibrationDimensions)
+                self.inputScales = newInputScales
+
+
+# CalibCombineConfig/CalibCombineTask from pipe_base/constructCalibs.py
+class CalibCombineConfig(pipeBase.PipelineTaskConfig,
+                         pipelineConnections=CalibCombineConnections):
+    """Configuration for combining calib exposures.
+    """
+    calibrationType = pexConfig.Field(
+        dtype=str,
+        default="calibration",
+        doc="Name of calibration to be generated.",
+    )
+    calibrationDimensions = pexConfig.ListField(
+        dtype=str,
+        default=[],
+        doc="List of updated dimensions to append to output."
+    )
+
+    exposureScaling = pexConfig.ChoiceField(
+        dtype=str,
+        allowed={
+            "None": "No scaling used.",
+            "ExposureTime": "Scale inputs by their exposure time.",
+            "DarkTime": "Scale inputs by their dark time.",
+            "MeanStats": "Scale inputs based on their mean values.",
+            "InputList": "Scale inputs based on a list of values.",
+        },
+        default=None,
+        doc="Scaling to be applied to each input exposure.",
+    )
+    scalingLevel = pexConfig.ChoiceField(
+        dtype=str,
+        allowed={
+            "Amplifier": "Scale by amplifier.",
+            "Detector": "Scale by detector.",
+        },
+        default="Detector",
+        doc="Region to scale.",
+    )
+    outputScaling = pexConfig.Field(
+        dtype=float,
+        default=float('nan'),
+        doc="Scaling value to apply to the output combined product.",
+    )
+
+    mask = pexConfig.ListField(
+        dtype=str,
+        default=["SAT", "DETECTED", "INTRP"],
+        doc="Mask planes to respect",
+    )
+    combine = pexConfig.Field(
+        dtype=int,
+        default=int(afwMath.MEANCLIP),
+        doc="Statistic to use for combination (from lsst.afw.math)",
+    )
+    clip = pexConfig.Field(
+        dtype=float,
+        default=3.0,
+        doc="Clipping threshold for combination",
+    )
+    nIter = pexConfig.Field(
+        dtype=int,
+        default=3,
+        doc="Clipping iterations for combination",
+    )
+    stats = pexConfig.ConfigurableField(
+        target=CalibStatsTask,
+        doc="Background statistics configuration",
+    )
+
+
+class CalibCombineTask(pipeBase.PipelineTask,
+                       pipeBase.CmdLineTask):
+    """Task to combine calib exposures."""
+    ConfigClass = CalibCombineConfig
+    _DefaultName = 'cpCombine'
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.makeSubtask("stats")
+
+    def runQuantum(self, butlerQC, inputRefs, outputRefs):
+        inputs = butlerQC.get(inputRefs)
```

Comment:
DM-22521 will implement partial image reads to avoid memory issues.  The short summary is that the inputs will be loaded as "deferred" objects, and the combination will iterate over subregions, loading only the active subregion from each deferred image.  I think this has been tested with regular image coadds.